File: Strings/Count and Say/python/count_say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        def say(value, n):
            alphabet = ""
            unique = value[0]
            count = 0
            for i in range(len(value)):
                if value[i] == unique:
                    count += 1
                else:
                    alphabet = alphabet + str(count) + unique
                    unique = value[i]
                    count = 1
            alphabet = alphabet + str(count) + unique
            if n == 0:
                return alphabet
            else:
                return say(alphabet, n - 1)

        if n == 1:
            return '1'
        else:
            return say("1", n - 2)


if __name__ == "__main__":
    test_n = 30
    #print(f"result of {test_n} is {Solution().countAndSay(test_n)}")
    print(f"length max = {len(Solution().countAndSay(test_n))}")

File: Easy/Strings/Count and Say/JS/count_say.js
/**
 * @param {number} result
 * @return {string}
 */
 var countAndSay = function(n) {
    

    var recursive_say = function(result, number){
        
        let unique_value = result[0];
        let count = 0;
        ret_result = ""
        /* 111223334 */
        for (let i of result){
            if (i === unique_value){
                count++;
            }
            else{
                ret_result = ret_result + count.toString() + unique_value;
                unique_value = i;
                count = 1;
            }
        }
        ret_result = ret_result + count.toString() + unique_value;

        if (number === 0){
            return ret_result;
        }
        else{
            return recursive_say(ret_result, number - 1);
        }
    }

    if (n === 1){
        return "1";
    }
    else{
        return recursive_say("1", n - 2);
    }
 }

 let test_n = 9;
 console.log(`result of say function is ${countAndSay(test_n)}`)